core/common/src/main/java/alluxio/grpc/GrpcServer.java

```diff
@@ -83,18 +94,23 @@ public int getBindPort() {
    * @throws InterruptedException
    */
   public boolean shutdown() {
+    // Stop accepting new connections.
     mServer.shutdown();
-    // Release authentication sessions if the server was authenticated.
-    if (mAuthServer != null) {
-      mAuthServer.close();
+    // Close resources that potentially owns active streams.
+    try {
+      mCloser.close();
+    } catch (IOException e) {
+      LOG.warn("Failed to close resources during shutdown.", e);
+      // Do nothing.
     }
+    // Force shutdown remaining calls.
+    mServer.shutdownNow();
```

Comment:
@gpang Currently, 1.8 branch interrupt and wait as per previous discussions. This is to make master branch comply with that decision as well.

Reply:
I am fine with interrupt and wait. But I would argue from user perspective to have a brief grace period before interruption to give existing RPC call some time to finish up.

Reply:
Then we'd have to have 2 waits periods. 1 graceful and 1 after interrupt.

Reply:
Yeah the graceful one doesn't have to be very long since most of the RPC calls shouldn't take more than a few seconds to complete.

Reply:
Having two separate waits (3 after counting executor wait) will complicate the code IMO.

Reply:
@gpang Do you have a preference here?

Reply:
I think `shutdown()` gets called as part of a master stepping down. In that case, we should not wait for anything, since we are required to stop serving as soon as possible, for correctness. Therefore, we should not wait for RPCs to finish. Also, any recursive RPC can take hours...

Reply:
We still need to think more about this area. Creating a ticket to track it.
#10505

Reply:
Talked to @ggezer offline. I am fine interrupt immediately as a quick fix to reduce leadership transition issues. Once those issues are addressed from their root cause, we should revisit the shutdown workflow and make the leadership transition less impactful to the users.